lib/core/test/unit/Cardano/Wallet/Api/ServerSpec.hs

```diff
@@ -166,7 +225,7 @@ networkInfoSpec = describe "getNetworkInformation" $ do
     it "doesn't return 500 when the time interpreter horizon is behind\
        \ the current time" $ property $ \(gap' ::(NonNegative Int)) ->
         monadicIO $ do
-        let gap = fromRational $ toRational $ getNonNegative gap'
+        let (gap :: NominalDiffTime) = fromRational $ toRational $ getNonNegative gap'
```

Comment:
OK, but isn't this the more normal way to write it?
```suggestion
        let gap = fromRational $ toRational $ getNonNegative gap' :: NominalDiffTime
```

Reply:
> but isn't this the more normal way to write it?

I have no idea, is it?

But to my eyes:
1. I think it reads more natural / I don't have a sense for the precedence of the `::`
2. When using a monadic bind, you'd also have to specify the type of the monad, which you might not want to:
```haskell
a <- getEnv "CI" :: IO String

:set -XScopedTypeVariables
a :: String <- getEnv "CI" 
```

Reply:
Well, that syntax requires the `ScopedTypedVariables` extension, so it's less normal than the normal way.
All you need to remember about `::` type annotations is that no operator has higher precedence.
The extension syntax is useful for `<-` but this here is a let expression.
There's nothing incorrect about it, but it just looked strange to me.